### PR TITLE
Convert video to GIF improvements

### PR DESCRIPTION
**Pull request by delphym**
Hi matey,

Great repository, especially with the MOV 2 GIF conversations.
Which I found very valuable, but I got frustrated with a couple of things:
- Apple Shortcuts – didn't work for me :( it's probably new thing, which is not available to me
- I replace it with [Automator workflow](https://support.apple.com/en-nz/guide/automator/aut7cac58839/mac), a bit of learning here for me. 
So now, everyone can invoke the conversion via a context menu  on a video file in `Finder` > `Quick Actions` (if they read the instructions)
- made the script a bit robust with an UI feedback to the user.
- updated the on-demand invocation with the `curl`

Feel free to take some bits from my PR or nothing. (The whole is not recommended because of the PATHes references.) So, it is not the PR that you could directly take and merge, but rather a communication conveyed to you.

### CODE REVIEW

**Comment by friedrith** [comment]
Hi! Thx a lot for your PR. I will take a deeper look during this weekend. 

Can you describe me what didn't go well with Apple shortcuts?

**Comment by delphym** [comment]
> Hi! Thx a lot for your PR. I will take a deeper look during this weekend.
You're welcome:) I'm glad you're responded.
You may like the shell script improvement, but maybe not the readme.  I guess, if it is working for the latest macOS, then you certainly want to keep the "Shortcuts" part. So, I might update it to have those two options of installations: via **Shortcuts** or via **Automator**. 
But I'll wait for your response.

> Can you describe me what didn't go well with Apple shortcuts?
Alright, I do my best: It literally did nothing when clicked on "Get Shortcut" button.
In the console I could see this error msg:
`Failed to launch 'workflow://shortcuts/db18892750f3453c83d57281915be17f' because the scheme does not have a registered handler.`
I guess it is because I don't have the Shortcuts app installed, but as I said it is not available to me :-( I'm on v11.7.10 and can't upgrade.
Screen Recording 2024-09-29 at 14 46 57

**Comment by friedrith** [comment]
It is weird. This shortcut works very well for on MacOS. Which version of MacOS do you have?

image

I checked more in details your changes and I will definitely merge them! I just want to test the shell script separately since it impacts both Apple Automator and my Apple Shortcut and I need to be sure it works for both.

**Comment by delphym** [comment]
Of course! Testing is important and I'm glad you've done it too:)
I spent a couple of days (not entirely) by improving and testing the script and then writing the doc (readme) as I thought it would be great to share it with the world but wanted to contribute the improvements back to the source:)

Well, as I wrote, I'm on v11.7.10 which is Big Sur and I can't upgrade, which I think is the reason, why it doesn't work for me.

I can revert (or improve the readme) instruction, so it will keep your original Apple Shortcut assets (as you've confirmed it is working) and I add my Aotmater bits there as a fallback instruction.


Let me know your thoughts.

**Comment by friedrith** [comment]
Yes I think it would a good idea to show the Automator as a fallback since it requires more setup. Apple Shortcuts just requires to click on the link. Maybe we can show the section like MacOS v... and below so people understand for which reason they should use Automator.

**Comment by delphym** [comment]
Oki, sounds like a plan. I'll update the instructions over the weekend (busy now) and yeah, hopefully, you could take those bits.

Cheers.

**Comment by friedrith** [comment]
> Oki, sounds like a plan. I'll update the instructions over the weekend (busy now) and yeah, hopefully, you could take those bits.
> 
> Cheers.

Thank you. A really appreciate!

**Comment by delphym** [comment]
Hi there, sorry it took me that long. Last weekend I was mostly away, so didn't have a chance.

Anyway, I did my best to update the [instructions](https://github.com/delphym/productivity/blob/master/convert-video-to-gif.md), which you can read nicely rendered in my repo.

The only concern I do have, is the *path* to the script as I use my repo in the path, instead of yours.... Not sure how to go around it.

I will highlight it in that *diff* for the instructions files.

File: convert-video-to-gif.md [hunk 1]
```diff
@@ -18,21 +18,78 @@ brew install ffmpeg imagemagick
...
+As mentioned earlier, you can opt in to use this tool via [script](#script) bellow.
 
 ### Script
 
 You can also run the script directly:
 
 ```bash
-curl https://raw.githubusercontent.com/friedrith/productivity/master/convert-video-to-gif.sh -o convert-video-to-gif.sh
-./convert-video-to-gif.sh <video-filename>
+curl -fsSL https://raw.githubusercontent.com/delphym/productivity/master/convert-video-to-gif.sh | zsh -s -- <video-2-convert.mov>
```
Comment (delphym) [inline convert-video-to-gif.md:79]: You may want this URL to point to your repo, I guess.

Comment (delphym) [reply convert-video-to-gif.md:79]: You probably need to do it manually after the merge as I cannot figure out a more convenient way.

**Comment by friedrith** [comment]
Thank you a lot for your PR, I also accepted the tag hacktoberfest so it can be added to your profile I think: https://hacktoberfest.com/

**Pull request merged**